File: alg_leetcode/495. Teemo Attacking.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int period = 0;
        int re_len = timeSeries.length-1;
        for (int i = 0;i < re_len;i=i+1){
            int lenth = timeSeries[i+1]-timeSeries[i];
            if (lenth < duration) {
                period = period + lenth;
            }else{
                period = period + duration;
            }
        }
        period = period + duration;
        return period;
        
    }
}

File: alg_leetcode/495. Teemo Attacking.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        period = 0
        for i in range(len(timeSeries)-1):
            if timeSeries[i+1] - timeSeries[i] < duration:
                period = period + timeSeries[i+1] - timeSeries[i]
            else:
                period = period + duration
        period = period + duration
        return period
